lib/volcanic/authenticator/v1/role.rb

```diff
@@ -63,15 +63,23 @@ def privilege_ids
         @privilege_ids.freeze
       end
 
-      def add_privilege(value)
-        dirty[:privilege_ids] = true
+      def add_privileges(*values)
```

Comment:
You've removed setting `dirty[:privilege_ids] = true`. This breaks the dirty checking.

You've also renamed the method. This breaks the interface and code that is calling it.

To fix:
* re-add the dirty setting
* add a new method `add_privilege(value)` which calls this with just the value

Reply:
@danleyden seems like `add_privileges` is using different path and it triggering the api to update the role for given privileges, by that im wondering if we can have it something like `add_privileges!(value)` where it can tell that this method will save to auth service immediately

Reply:
a different method is fine - just need to be wary that the "dirty" flag is still adhered to, otherwise the caching breaks and a lot of things will get a LOT slower or return incorrect results